packages/core/src/models/tool-messages.ts

```diff
@@ -34,6 +34,8 @@ export const ToolResponseSchema = z.object({
   type: z.literal('tool_response'),
   tool_call_id: z.string(),
   content: ToolResponseContentSchema,
+  tool_name: z.string().optional(),
+  tool_type: z.enum(['builtin', 'function']).optional(),
```

Comment:
it might be better to use union + literal here instead of enum

```suggestion
  tool_type: z.union([z.literal('builtin'), z.literal('function')]).optional(),
```